src/reference/modules/reference/pages/relay_api.adoc

```diff
@@ -1,102 +1,191 @@
+There are 3 types of HTTP responses:
 
-There are ncf generic method that allow easy access to those methods from the nodes.
+* HTTP 404 : the target-uuid is unknown
+* HTTP 500 : a problem occurred (unable to get the nodes list, no ttl
+provided, unable to get file content)
+* HTTP 200 : everything went well
+
+Once the file has been forwarded to the right node, a metadata file is
+saved next to the shared file. It contains the _PUT_ parameters sent
+with the file. Here’s an example of a such file:
+
+`+header=rudder-signature-v1+` `+algorithm=sha512+`
+`+digest=8ca9efc5752e133e2[..]CAwEAAQ==+` `+hostname=ubuntu-18-04-64+`
+`+keydate=2018-10-31 18:21:43.653257143 +0000+` `+keyid=B29D02BB+`
+`+expires=1562230419+`
+
+NB : the `+expires+` field contains the ttl parameter sent, in a unix
+timestamp format.
+
+== HEAD
+
+The URL parameters specific to the _HEAD_ call are:
+
+* `+file-hash+` = _value of the hash_: hash of the shared file
+
+The purpose of the HEAD request is to check if the file named
+`+<file-id>+` stored in `+/shared-files/<target-uuid>/<source-uuid>/+`
+has its hash value equal to the value sent with the `+?hash=+`
+parameter.
+
+The relay that receives a _HEAD_ call will respond with:
+
+* HTTP 200 if the hash values are equal
+* HTTP 404 if the hash values are not equal, or the target-uuid is
+unknown
+* HTTP 500 : a problem occurred (unable to get the nodes list, no ttl
+provided, unable to get file content)
+
+== Shared Folder
+
+The goal of this API is to share a folder between Windows nodes.
+
+=== Description
 
 === Security
 
-This call is open to all nodes in the allowed networks of the target relay.
-The sent files are signed with the node's key, and the signature is checked before being traited.
+This call is open to all nodes in the allowed networks of the target
+relay. The sent files are signed with the node’s key, and the signature
+is checked before being processed.
 
 === Usage
 
 This API provides the following methods:
 
-* *PUT* `/shared-files/` _<target-uuid>_ `/` _<source-uuid>_ `/` _<file-id>_
-* *HEAD* `/shared-files/` _<target-uuid>_ `/` _<source-uuid>_ `/` _<file-id>_ `?hash=` _file-hash_
-
-The common URL parameters are:
-
-* `target-uuid` = *destination_node_uuid*:  where to send the file to
-* `source-uuid` = *source_node_uuid*: who sent the file
-* `file-id` = *my_file_id*: under which name to store the file, this needs to be unique
+* HEAD or GET
+`+/shared-folder/<path:file_name>?hash_type=<sha1 || sha256 || sha512>?hash=<hash_value>+`
 
-The URL parameters specific to the *HEAD* call are:
+=== HEAD & GET
 
-* `file-hash` = *value of the hash*: hash of the shared file
+* `+hash_type+` = _sha1_, _sha256_ or _sha512_: algorithm used to hash
+the file
+* `+hash_value+` = _value of the hash_: hash of the shared file
 
-The following are only needed for the *PUT* call:
+The relay that receives a _HEAD_ or a _GET_ call will respond with:
```

Comment:
I think the `GET` here is actually a bug, behavior seems exactly the same. We should remove it and just add a note saying it is now deprecated.